src/virtual-machine.js

```diff
@@ -289,6 +290,16 @@ class VirtualMachine extends EventEmitter {
         }
 
         const validationPromise = new Promise((resolve, reject) => {
+            try {
+                const sb1 = new SB1File(input);
+                const json = sb1.json;
+                json.projectVersion = 2;
+                console.log(json);
+                return resolve([json, sb1.zip]);
+            } catch (e) {
+                console.error(e);
```

Comment:
Errors should probably get thrown like the rest of this code does.

At any rate, even if we're logging the error, you should use `log.error` here.

Reply:
I switch to `log.error`.

The errors that would happen here are validation errors that would indicate the passed data is not an SB1 File. If we threw those non-SB1 projects would not be able to load anymore.

We could change this to split the error handling behaviour on whether the error is a sb1-converter AssertionError (with a small change to sb1-converter to export its AssertionError). (Probably in a follow up PR?)

Reply:
Yeah, it sounds like without the change you're suggesting, we will be constantly logging errors that aren't actually errors (e.g. the normal use cases of loading sb2 and sb3 files)...